packages/smooth_app/lib/pages/product/new_product_page.dart

```diff
@@ -219,33 +226,133 @@ class _ProductPageState extends State<NewProductPage> {
+      child: ListTile(
+        contentPadding: EdgeInsets.zero,
+        title: Text(
+          _getProductName(appLocalizations),
+          style: themeData.textTheme.headline4,
+        ),
+        subtitle: Text(_product.brands ?? appLocalizations.unknownBrand),
+        trailing: Text(
+          _product.quantity ?? '',
+          style: themeData.textTheme.headline3,
+        ),
+      ),
+    );
+  }
+
+  Widget getAttributeGroupWidget(
+      BuildContext context, Product product, String groupId) {
+    final Iterable<AttributeGroup> groupIterable = _product.attributeGroups!
+        .where((AttributeGroup group) => group.id == groupId);
+    if (groupIterable.isEmpty) {
+      return emptyWidget_;
+    }
+    final AttributeGroup group = groupIterable.single;
+    return Container(
+      alignment: Alignment.topLeft,
+      child: Column(
+        children: <Widget>[
+          Container(
             alignment: Alignment.topLeft,
-            child: ListTile(
-              contentPadding: EdgeInsets.zero,
-              title: Text(
-                _getProductName(appLocalizations),
-                style: themeData.textTheme.headline4,
-              ),
-              subtitle: Text(_product.brands ?? appLocalizations.unknownBrand),
-              trailing: Text(
-                _product.quantity ?? '',
-                style: themeData.textTheme.headline3,
-              ),
+            padding: const EdgeInsets.symmetric(vertical: 8),
+            child: getAttributeGroupHeader(context, group),
+          ),
+          Container(
+            alignment: Alignment.topLeft,
+            child: Wrap(
+              runSpacing: 16,
+              children: <Widget>[
+                for (final Attribute attribute in group.attributes!)
+                  getAttributeChipForValidAttributes(context, attribute) ??
+                      emptyWidget_,
+              ],
             ),
           ),
-          for (final Attribute attribute in scoreAttributes)
-            ScoreAttributeCard(attribute: attribute, iconHeight: iconHeight),
-          attributesContainer
-        ]),
+        ],
       ),
     );
-    return ListView(children: listItems);
   }
 
-  Widget? getAttributeChip(Attribute attribute, Size screenSize) {
+  /// The attribute group header can either be group name or a divider depending
+  /// upon the type of the group.
+  Widget getAttributeGroupHeader(BuildContext context, AttributeGroup group) {
+    final ProductPreferences productPreferences =
+        context.watch<ProductPreferences>();
+    final bool containsImportantAttributes = group.attributes!.any(
+        (Attribute attribute) =>
+            productPreferences.isAttributeImportant(attribute.id!));
+    if (!containsImportantAttributes) {
+      return emptyWidget_;
+    }
+    if (group.id == AttributeGroup.ATTRIBUTE_GROUP_ALLERGENS) {
+      return
+        Padding(
+          padding: const EdgeInsets.only(bottom: 8),
+          child:Text(
```

Comment:
Bad formatting. Run `flutter format`?

Reply:
`child:Text(` is missing a space after the colon and the following line ought to be intended 2 spaces more than this line.

Reply:
ah in that case it must not have been flutter formatted